### ISSUE TITLE
Question: VSIX with Project and Item templates

### ISSUE THREAD

**Issue by ivanpovazan** [sayedihashimi/template-sample#53]
I have troubles with a VSIX file which includes project and item templates.
I was following your guide from README and tried to configure the extension to include both Items and Projects template.

1) I have two different packages:
```
ItemTemplates.nupkg
ProjectTemplates.nupkg
```
2) I included them in the VSIX with something like:
```
<ItemGroup>
    <VSIXSourceItem Include="$(ProjectTemplatesPackage)">
      <VSIXSubPath>ProjectTemplates\</VSIXSubPath>
    </VSIXSourceItem>
   <VSIXSourceItem Include="$(ItemTemplatesPackage)">
      <VSIXSubPath>ItemTemplates\</VSIXSubPath>
    </VSIXSourceItem>
</ItemGroup>
```
where ProjectTemplatesPackage and ItemTemplatesPackage project properties point to respective .nupkgs

3) In the Templates.pkgdef
I have added an additional line:
```
[$RootKey$\TemplateEngine\Templates\ProjectTemplates\1.0.0]
"InstalledPath"="$PackageFolder$\ProjectTemplates"
[$RootKey$\TemplateEngine\Templates\ItemTemplates\1.0.0]
"InstalledPath"="$PackageFolder$\ItemTemplates"
```

4) I made sure that once .vsix is made in includes \ItemTemplates and \ProjectTemplates folder (when unzipped) and in those appropriate nuget packages

However, once I install the .vsix the project templates are recognized when "New Project" is chosen, but on "New Item" I cannot seem to find the Item templates. Is there something I am missing, or a known issue?

**Comment by sayedihashimi** [comment]
Hi @ivanpovazan Item templates are not currently supported. We are working with the team that owns dotnet new to get some functionality added so that we can support Item Templates.

**Comment by ivanpovazan** [comment]
Hi @sayedihashimi, 
Thank you for the update. Any estimate on when will this be available?

**Comment by sayedihashimi** [comment]
@ivanpovazan probably not for a while. If we are able to work out the changes needed in `dotnet new`, those would likely go into .net 7. So some point after that is the earliest.

There is a way to get item templates to work, but it's ***not supported***. I think you can get it to work, but when we change to the new method it will stop working. I want to be clear that if you get it working, and run into issues, we may not fix those issues, and instead wait for the long term approach. If you are interested in exploring this unsupported path, let me know and I will give you some info on how you can try it out.

**Comment by ivanpovazan** [comment]
I see.
To be honest, I would be really interested to explore the "unsupported" path :)
As a backup I have already left Item templates to work with the 'old' vstemplate approach, but would like to experiment.

**Comment by sayedihashimi** [comment]
# The following is not supported and may break and not get fixed

@ivanpovazan here is the info on how you can surface your Template Engine Item Templates in VS2022. Reminder that this will likely stop working in the future and you'll have to migrate to a new method in the future.

To get Item Templates to appear, you'll need to add a file named `ide.host.json` in the `.template.config` folder.

You can see a sample of an item template at https://github.com/dotnet/maui/blob/main/src/Templates/src/templates/maui-contentpage-xaml/.template.config/ide.host.json. Let me break down the contents of that file.

```json
{
    "$schema": "http://json.schemastore.org/vs-2017.3.host",
    "appliesTo": "UseMaui",
    "defaultItemExtension": "xaml",
    "itemHierarchyPaths": [ ".NET MAUI" ],
    "icon": "ide/MauiPageXamlFile.ico",
    "unsupportedHosts": [
        {
            "id": "vs",
            "version": "(,17.1)"
        }
    ]
}
```

`appliesTo` this is an expression that limits when this template will appear. If you want your template to appear for all projects, you can leave this off. To make this appear for certain types of projects you'll need to add the name of a Project Capability in `appliesTo`. To get the list of Project Capabilities that a project type supports, add the following to your project file.

```xml
<ItemGroup>
    <ProjectCapability Include="DiagnoseCapabilities"/>
</ItemGroup>
```

After adding this, you should see a node in the Solution Explorer under that project that shows the project capabilities. Identify a specific capability that you want to condition on an add it to `appliesTo`.

`defaultItemExtension` default extension for the file being created. For ex, if your template creates a CSS file, this value should be set to "css".

`itemHierarchyPaths` node path to where this template should be shown in the New Item Dialog in VS2022. In this example the template is being shows in the top level ".NET MAUI" node. To show the template in a sub-node use `\` as the separeator. For example `Path2\SubPath`

`icon` path to the icon that should be shown for this item template in the New Item Dialog. The icon file must be in, or under, the `.template.config` folder.

`unsupportedHosts` copy/paste that as is.

Please give it a try and let me know if you have questions.

**Comment by ivanpovazan** [comment]
Thank you for providing this example.
Is this approach limited to VS 17.1 preview since I see that the item templates are hidden for lower versions?

**Comment by sayedihashimi** [comment]
> Is this approach limited to VS 17.1 preview

Yes, we recently implemented that support. Sorry I forgot to mention that.

**Comment by ivanpovazan** [comment]
Unfortunately, I did not manage to make it work even with VS 17.1. 
For now, I will keep the .vstemplate files for the VSIX and for the item template package I have add template.jsons on the same folder structure with a bit of additional configuration.
All in all, I have managed to make the templates from VSIX and from template package generate same results.

Will wait for .net7 to simplify this process.
Thank you
